cmake_modules/IcebergSanitizer.cmake

```
@@ -0,0 +1,46 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Sanitize check for address and undefined.
+
+if(ICEBERG_ENABLE_ASAN OR ICEBERG_ENABLE_UBSAN)
+  add_library(iceberg_sanitizer_flags INTERFACE)
+  set(SANITIZER_FLAGS iceberg_sanitizer_flags)
+endif()
```

```suggestion
add_library(iceberg_sanitizer_flags INTERFACE)
```

It is much simpler to make it available at all times even without any flag.